### DISCUSSION TITLE
Canonical, meta-og:url, meta-og:image with getStaticProps?

### DISCUSSION THREAD

**Discussion by aralroca** [vercel/next.js#11856] [Help] [answered]
First of all, I want to say how impressive/easy is to use `getStaticProps` to build a blog, reading directly `.md` files! 👏 I'm currently redoing my blog to Next.js SSG: https://github.com/aralroca/aralroca.com 

However, I'm getting troubles adding SEO info as `canonical`, `meta-og:url`, `meta-og:image`. The reason is that they required an absolute URL, and in "build" moment I don't know the hostname... 

I'm using Zeit as hosting. Is there a default environment variable or some way to know the current domain?

Thank you so much!

**Comment by tyteen4a03** [comment]
Bump

**Comment by IanMitchell** [comment, marked as answer]
Try reading the environmental variables during build: https://vercel.com/docs/v2/build-step#system-environment-variables

**Comment by tyteen4a03** [comment]
BTW For me I found [next-seo](https://github.com/garmeeh/next-seo) which solves my use case.
